### PR TITLE
Improve websocket logging

### PR DESCRIPTION
**Pull request by agnivade**
After reliable websockets were introduced,
we would keep pushing to a webConn even after
the client has disconnected. This would lead
to the websocket.slow/full and eventually
the disconnect messages which unnecessarily
confuse the admin.

We don't log them if the connection is inactive.

We use the active field and convert it to an atomic
to use more widely.

```release-note
NONE
```

### CODE REVIEW

**Comment by agnivade** [comment]
@amyblais - I'd want to cp this to the last ESR. It's an important change that has an impact on reducing misleading warning logs.

File: server/channels/app/platform/web_hub.go [hunk 1]
```diff
@@ -478,9 +478,12 @@ func (h *Hub) Start() {
...
 				default:
-					mlog.Error("webhub.broadcast: cannot send, closing websocket for user",
-						mlog.String("user_id", directMsg.conn.UserId),
-						mlog.String("conn_id", directMsg.conn.GetConnectionID()))
+					// Don't log the warning if it's an inactive connection.
+					if directMsg.conn.active.Load() {
+						mlog.Error("webhub.broadcast: cannot send, closing websocket for user",
+							mlog.String("user_id", directMsg.conn.UserId),
+							mlog.String("conn_id", directMsg.conn.GetConnectionID()))
+					}
```
Comment (streamer45) [inline server/channels/app/platform/web_hub.go:486]: Wondering if we should just drop the level in inactive (e.g. debug). I am just worried that completely losing the log may hinder future debugging due to decreased visibility on what's happening.

**Review by streamer45 — APPROVED** [review]
Thanks, only left one minor thought.

File: server/channels/app/platform/web_hub.go (hunk 1 above)
Comment (agnivade) [reply server/channels/app/platform/web_hub.go:486]: Hmm .. ok. Maybe we could also change the text slightly to indicate that it's an inactive connection.

**Review by isacikgoz — APPROVED** [review]
LGTM, 1/5 on logging debug level if the connection is inactive, but we can have error logs for the active conn.

File: server/channels/app/platform/web_hub.go (hunk 1 above)
Comment (agnivade) [reply server/channels/app/platform/web_hub.go:486]: On a deeper analysis, I recalled that we already have Prom metrics that capture websocket reconnects by their type, and also the websocket broadcast buffer gauge. Combining both of these, we should be able to detect if there are too many failed reconnects happening or whether the system is choking as a whole.

Given that before the reliable websockets feature, we would just drop the connection and not log anything at all, this PR would effectively preserve original behavior.

There's also the fact that a number of customers and community users are a bit confused with these extra logs and think that something is wrong with the system. As a first step, I would like to be on the safe side and just try to reduce noise. If, even with the Prom metrics, we feel a need to capture the inactive disconnections, then we can add that. But I would try to refrain from preemptively adding them at a debug level.

**Comment by agnivade** [comment]
@amyblais - Should I cp to release-7.8 now?

**Pull request merged**